src/rust/engine/grpc_util/src/lib.rs

```diff
@@ -144,6 +155,34 @@ pub fn headers_to_metadata_map(headers: &BTreeMap<String, String>) -> Result<Met
   Ok(metadata_map)
 }
 
+pub fn headers_to_http_header_map(headers: &BTreeMap<String, String>) -> Result<HeaderMap, String> {
+  let http_headers = headers
+    .iter()
+    .map(|(key, value)| {
+      let header_name = HeaderName::from_str(key.as_str())
```

Comment:
Feel free to ignore. I personally prefer this style because it feels more consistent with the rest of the language, like how I would do this for a non-String type.

```suggestion
      let header_name = HeaderName::from_str(&key)
```

Reply:
Key is `&String` in this closure. So I don't believe it would dereference to `&str`.

Reply:
FWIW [implicit deref coercion](https://doc.rust-lang.org/book/ch15-02-deref.html#implicit-deref-coercions-with-functions-and-methods) means Eric's suggestion _should_ still work :)